### PR TITLE
Feature: AI Node and AI Actions extension (#26055) from symphony

### PR DESCRIPTION
**Pull request by fmontes**
* Created AI content node

* Created ai-content-service

* Created AI prompt content extension

* Update ai-content.service.ts

* Optimised and aligned extension code

* Resolved comments on PR

* Removed ai content node file

* Update ai-content.service.ts

* Remove import path for ai-content-node

* Updated the ai-prompt form

* align ai text prompt extension

* Resovled comments on pull

* Fix the aiTextPrompt form

* Updated extension related code

* Update ai-content-prompt.plugin.ts

* Remove some extra code

* Fixed outside click for aiContentPrompt extension

* Created ai content node

* Updated ai-content node

* Remove unused code, resolve comments on PR

* ai-content-prompt extension code optimisation

* Update ai-content-prompt.plugin.ts

* Integrated with AI api

* insert ai node on response

* textPrompt extension code optimsation

* Update ai-content.service.ts

* Resolve comments on pull req

* handle close extension on outside click and content flip fix

* Added focus field method and type for form

* Change name of ai node creation command

* Update the ai-content icon in the actions menu list

* Update ai-content-prompt.component.scss

* Resolved comments on pull

* Added pening status and update the name of destroy var

* Implement ai-content-actions extension

* ai-prompt extension code optimisation

* ai-text-prompt block optimisation

* Update main.ts

* Resolve comments on pull

* Resolve comments on PR

* Code optimisation for ai-content-actions extension

* Fix css for p-listbox

---------

### Proposed Changes
* change 1
* change 2

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

**Review by rjvelazco — APPROVED** [review]
We should implement a feature flag for this AI Extension before building the `dotcms-block-editor` because this extension hasn't passed to QA.

File: core-web/libs/block-editor/src/lib/extensions/ai-content-actions/ai-content-actions.component.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    @Output() actionEmitter = new EventEmitter<ACTIONS>();
+
+    actionOptions!: ActionOption[];
+
+    constructor(private aiContentService: AiContentService) {}
+
+    ngOnInit() {
+        this.actionOptions = [
+            {
+                label: 'Accept',
```
Comment (oidacra) [inline core-web/libs/block-editor/src/lib/extensions/ai-content-actions/ai-content-actions.component.ts:35]: Translation

File: core-web/libs/block-editor/src/lib/extensions/ai-content-actions/ai-content-actions.component.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                icon: 'pi pi-check',
+                callback: () => this.emitAction(ACTIONS.ACCEPT),
+                selectedOption: true
+            },
+            {
+                label: 'Regenerate',
```
Comment (oidacra) [inline core-web/libs/block-editor/src/lib/extensions/ai-content-actions/ai-content-actions.component.ts:41]: Translation

File: core-web/libs/block-editor/src/lib/extensions/ai-content-actions/ai-content-actions.component.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                icon: 'pi pi-sync',
+                callback: () => this.emitAction(ACTIONS.REGENERATE),
+                selectedOption: false
+            },
+            {
+                label: 'Delete',
```
Comment (oidacra) [inline core-web/libs/block-editor/src/lib/extensions/ai-content-actions/ai-content-actions.component.ts:47]: Translation

File: core-web/libs/block-editor/src/lib/extensions/ai-content-actions/ai-content-actions.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ReactiveFormsModule } from '@angular/forms';
+
+import { AIContentActionsComponent } from './ai-content-actions.component';
+
+import { AiContentService } from '../../shared';
+
+describe('AIContentActionsComponent', () => {
+    let component: AIContentActionsComponent;
```
Comment (oidacra) [inline core-web/libs/block-editor/src/lib/extensions/ai-content-actions/ai-content-actions.component.spec.ts:10]: Add more test and convert to spectator

File: core-web/libs/block-editor/src/lib/extensions/ai-content-actions/ai-content-actions.component.scss [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+            // accept option
+            &:first-child {
+                background-color: $color-palette-primary-200;
+            }
+
+            // regenerate option
+            &:nth-child(2) {
+                padding: $spacing-3;
+                background-color: $white !important;
```
Comment (oidacra) [inline core-web/libs/block-editor/src/lib/extensions/ai-content-actions/ai-content-actions.component.scss:21]: Find the way to not use !important

File: core-web/libs/block-editor/src/lib/extensions/ai-content-actions/ai-content-actions.component.scss [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+            }
+
+            // delete option
+            &:last-child {
+                border-top: 1px solid $color-palette-gray-300;
+                background-color: $white !important;
```
Comment (oidacra) [inline core-web/libs/block-editor/src/lib/extensions/ai-content-actions/ai-content-actions.component.scss:27]: Find the way to not use !important

File: core-web/libs/block-editor/src/lib/extensions/ai-content-actions/ai-content-actions.component.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+export enum ACTIONS {
+    ACCEPT = 'ACCEPT',
+    DELETE = 'DELETE',
+    REGENERATE = 'REGENERATE'
+}
+
+@Component({
+    selector: 'dot-ai-content-actions',
+    templateUrl: './ai-content-actions.component.html',
+    styleUrls: ['./ai-content-actions.component.scss']
```
Comment (oidacra) [inline core-web/libs/block-editor/src/lib/extensions/ai-content-actions/ai-content-actions.component.ts:23]: add OnPush

File: core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/ai-content-prompt.component.ts [hunk 8]
```diff
@@ -21,6 +21,7 @@ export class AIContentPromptComponent {
 
     @ViewChild('input') private input: ElementRef;
     @Output() formSubmission = new EventEmitter<boolean>();
+    @Output() aiResponse = new EventEmitter<string>();
```
Comment (oidacra) [inline core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/ai-content-prompt.component.ts:24]: add OnPush

**Review by oidacra — APPROVED** [review]

**Pull request merged**